apps/api/src/domain/models/digital-text/commands/create-digital-text.command-handler.ts

```diff
@@ -0,0 +1,146 @@
+import getInstanceFactoryForResource from '../../../factories/getInstanceFactoryForResource';
+import { EVENT, IIdManager } from '../../../interfaces/id-manager.interface';
+import { IRepositoryForAggregate } from '../../../repositories/interfaces/repository-for-aggregate.interface';
+import { IRepositoryProvider } from '../../../repositories/interfaces/repository-provider.interface';
+import { AggregateId } from '../../../types/AggregateId';
+import { InMemorySnapshot, ResourceType } from '../../../types/ResourceType';
+import buildInMemorySnapshot from '../../../utilities/buildInMemorySnapshot';
+import { BaseCommandHandler } from '../../shared/command-handlers/base-command-handler';
+import UuidNotGeneratedInternallyError from '../../shared/common-command-errors/UuidNotGeneratedInternallyError';
+import { BaseEvent } from '../../shared/events/base-event.entity';
+import { DigitalText } from '../digital-text.entity';
+import { CreateDigitalText } from './create-digital-text.command';
+import { DigitalTextCreated } from './digital-text-created.event';
+
+@CommandHandler(CreateDigitalText)
+export class CreateDigitalTextCommandHandler extends BaseCommandHandler<DigitalText> {
+    protected repositoryForCommandsTargetAggregate: IRepositoryForAggregate<DigitalText>;
+
+    constructor(
+        @Inject(REPOSITORY_PROVIDER_TOKEN)
+        protected readonly repositoryProvider: IRepositoryProvider,
+        @Inject('ID_MANAGER') protected readonly idManager: IIdManager
+    ) {
+        super(repositoryProvider, idManager);
+
+        this.repositoryForCommandsTargetAggregate =
+            this.repositoryProvider.forResource<DigitalText>(ResourceType.digitalText);
+    }
+
+    protected async createOrFetchWriteContext({
+        aggregateCompositeIdentifier: { id },
+        title,
+        languageCodeForTitle,
+    }: CreateDigitalText): Promise<ResultOrError<DigitalText>> {
+        const digitalTextDto: DTO<DigitalText> = {
+            id,
+            title: buildMultilingualTextWithSingleItem(title, languageCodeForTitle),
+            published: false,
+            type: ResourceType.digitalText,
+            eventHistory: [],
+        };
+
+        return getInstanceFactoryForResource<DigitalText>(ResourceType.digitalText)(digitalTextDto);
+    }
+
+    actOnInstance(digitalText: DigitalText): ResultOrError<DigitalText> {
+        return digitalText;
+    }
+
+    protected async fetchRequiredExternalState(): Promise<InMemorySnapshot> {
+        const searchResult = await this.repositoryProvider
+            .forResource<DigitalText>(ResourceType.digitalText)
+            .fetchMany();
+
+        const allDigitalTexts = searchResult.filter((digitalText): digitalText is DigitalText => {
+            if (isInternalError(digitalText)) {
+                throw digitalText;
+            }
+
+            return true;
+        });
+
+        // Deal with deprecation? Use DeluxeInMemoryStore?
```

Comment:
I'm not sure if this is still an issue when you inherit from the `BaseCreateCommandHandler`. If so, you want to 
```ts
return new DeluxeInMemoryStore({
            resources: { digitalText: allDigitalTexts },
        }).fetchFullSnapshotInLegacyFormat()
```
It's on my TODO list to phase out the old format and rename these calls to `fetchFullSnapshot` or something like that.

To add some context here, the point is that I realized that I made a mistake by nesting resources on the snapshot as 
```ts
snapshot.resources.term
```
instead of 
```ts
snapshot.term
```
and I wanted to use the flattened structure going forward without updating existing APIs for helpers or existing usages. But it's gone pretty far. I need to clean this up soon.